### PR TITLE
Limit the visibility of a compliance app in store by role(permissions…

### PR DESCRIPTION
**Pull request by davidpofo**
…) and individual.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,6 +60,8 @@ jobs:
     # vulnerabilities using pyup.io, and run the check.
     - run: pip3 install --user -U safety
     - run: safety check --bare -r requirements.txt
+    - run: python3 manage.py makemigrations
```
Comment (gregelin) [inline .circleci/config.yml:63]: Migrations should already be committed to code base and results tested before any deployments, so `makemigrations` should not appear in any deployment processes.

**Review by gregelin — CHANGES_REQUESTED** [review]
I think we need tests written to test permission rules.

**Comment by gregelin** [comment]
@davidpofo This is nice work. Thank you for digging into backlogged issues to tackle. I think we want to add tests for rule set before committing.

File: .circleci/config.yml (hunk 1 above)
Comment (davidpofo) [reply .circleci/config.yml:63]: Yes noticed my local migration files weren't being committed so reverting this.

**Review by peterkaminski — APPROVED** [review]
reviewed with David, lgtm

**Comment by gregelin** [comment]
@davidpofo @peterkaminski I've just verified that the code change in this branch is breaking tests in `siteapp.tests.QuestionsTests`.  Can you please correct the tests to adjust for the new filtering of the available apps?

A variety of Selenium tests involve instantiating apps from the apps catalog. This is especially true with `siteapp.tests.QuestionsTests` and with discussion tests. 

When the change was made to limit who could see the apps in the apps store, the related tests did not get the memo and have their code adjusted to be able to have permission to see the tests compliance apps that are set up for the tests.

The test: 
```
 python manage.py test siteapp.tests.QuestionsTests
# or
 python manage.py test
```

Test suite fail on these commits:

* 032e0ba4 - (HEAD -> master, origin/master, origin/HEAD) update google-font-download - fix for #1102 (#1105) (66 minutes ago) <Peter Kaminski>
* 12c19fa3 - Add organizational parameter value substitution.  (#1088) (6 hours ago) <Tom Wood>
* f487c1b5 - Limit the visibility of a compliance app in store by role(permissions… (#1065) (9 days ago) <davidpofo>


Tests pass on these commits:

* 643c8d3c - (ge/fix-multiline-string-oscal-ssp) Update jquery to 3.5.1, quill to 1.3.7, bootstrap to 3.4.1 (#1052) (13 days ago) <bama4>
* 79c76750 - Fixed loose file permissions for /tmp/environment.json file (#1055) (2 weeks ago) <bama4>
...

The tests fail because the necessary compliance apps no longer appear in the app store/catalog when Selenium goes to choose the app.

**Comment by davidpofo** [comment]
> @davidpofo @peterkaminski I've just verified that the code change in this branch is breaking tests in `siteapp.tests.QuestionsTests`. Can you please correct the tests to adjust for the new filtering of the available apps?
> 
> A variety of Selenium tests involve instantiating apps from the apps catalog. This is especially true with `siteapp.tests.QuestionsTests` and with discussion tests.
> 
> When the change was made to limit who could see the apps in the apps store, the related tests did not get the memo and have their code adjusted to be able to have permission to see the tests compliance apps that are set up for the tests.
> 
> The test:
> 
> ```
>  python manage.py test siteapp.tests.QuestionsTests
> # or
>  python manage.py test
> ```
> 
> Test suite fail on these commits:
> 
> * [032e0ba](https://github.com/GovReady/govready-q/commit/032e0ba4f2efe47b29f462e7529ac92e1d29e089) - (HEAD -> master, origin/master, origin/HEAD) update google-font-download - fix for #1102 (#1105) (66 minutes ago)
> * [12c19fa](https://github.com/GovReady/govready-q/commit/12c19fa3f4d1d9593cfebfbf8c595011b5b39819) - Add organizational parameter value substitution.  (#1088) (6 hours ago)
> * [f487c1b](https://github.com/GovReady/govready-q/commit/f487c1b5f31a38dda3b4c3d99a07c23f90dbc913) - Limit the visibility of a compliance app in store by role(permissions… (#1065) (9 days ago)
> 
> Tests pass on these commits:
> 
> * [643c8d3](https://github.com/GovReady/govready-q/commit/643c8d3c8bbffca6b3f16a8c06c9530ccfb4881f) - (ge/fix-multiline-string-oscal-ssp) Update jquery to 3.5.1, quill to 1.3.7, bootstrap to 3.4.1 (#1052) (13 days ago)
> * [79c7675](https://github.com/GovReady/govready-q/commit/79c767500e123328d2466562befb622804c9d202) - Fixed loose file permissions for /tmp/environment.json file (#1055) (2 weeks ago) 
>   ...
> 
> The tests fail because the necessary compliance apps no longer appear in the app store/catalog when Selenium goes to choose the app.

Addressed in [this PR ](https://github.com/GovReady/govready-q/pull/1112)

**Pull request merged**